engine/src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -266,18 +274,14 @@ case class JesBackend(actorSystem: ActorSystem)
   }
 
   /**
-   * Get a GcsLocalizing from workflow options if client secrets and refresh token are available.
-   */
-  def getGcsAuthInformation(workflow: WorkflowDescriptor): Option[JesAuthInformation] = {
-    def extractSecrets(gcsMode: GoogleAuthMode) = gcsMode match {
-      case RefreshTokenMode(secrets) => Option(secrets)
-      case _ => None
-    }
-
+    * Get a GcsLocalizing from workflow options iff client secrets and refresh token are available as this method
+    * is called only if an auth.json needs to be written.
+    */
+  def refreshTokenAuth(workflow: WorkflowDescriptor): Option[JesAuthInformation] = {
     for {
-      secrets <- extractSecrets(gcsConf.authMode)
-      token <- workflow.workflowOptions.get(GoogleConfiguration.RefreshTokenOptionKey).toOption
-    } yield GcsLocalizing(secrets, token)
+      clientSecrets <- Seq(jesAttributes.gcsFilesystemAuth) collectFirst { case s: ClientSecrets => s }
```

Comment:
I'll grant that is weird but is there something in particular that would make you 😺 ?